### DISCUSSION TITLE
Improvements for the GUI

### DISCUSSION THREAD

**Discussion by carlosedp** [chucker/Mastonaut#40] [Ideas]
Here are some ideas (that might or might not be possible due to API limitations):

- It would be great to have separate columns for Mentions and the rest (follows, favorites, etc)
- Have a GUI button or an input box for search instead of only the menu item and keyboard shortcut

**Comment by chucker** [comment]
>It would be great to have separate columns for Mentions and the rest (follows, favorites, etc)

This is possible (the API returns a "type" for each notification), but it does complicate the UI a little. I guess what I can see is having a submenu, where you can either choose Notifications directly and get the default setup, or go to the submenu to get one particular kind of notification.

> Have a GUI button or an input box for search instead of only the menu item and keyboard shortcut

Yes; see also https://github.com/chucker/Mastonaut/issues/12. I think I should simply add it to the toolbar.

**Reply by carlosedp** [reply]
> > It would be great to have separate columns for Mentions and the rest (follows, favorites, etc)
> 
> This is possible (the API returns a "type" for each notification), but it does complicate the UI a little. I guess what I can see is having a submenu, where you can either choose Notifications directly and get the default setup, or go to the submenu to get one particular kind of notification.
> 
This would be great, so one could add a new column and "filter" for the types needed (multiple selection if will implement all types).

> > Have a GUI button or an input box for search instead of only the menu item and keyboard shortcut
> 
> Yes; see also #12. I think I should simply add it to the toolbar.

Yes, having it in the toolbar would be nice!

**Comment by chucker** [comment]
Sooooo, maybe the notifications preferences could have a [ ] Treat Mentions as a separate column checkbox, and then Mentions would be disabled as a checkbox in the notification filters. Hmm.
